cpp/include/cudf/join.hpp

```diff
@@ -438,6 +446,82 @@ class hash_join {
   const std::unique_ptr<impl_type const> _impl;
 };
 
+/**
+ * @brief Unique hash join that builds hash table in creation and probes results in subsequent
+ * `*_join` member functions
+ *
+ * @note Behavior is undefined if the build table contains duplicates.
+ * @note All NaNs are considered as equal
+ *
+ * @tparam HasNested Flag indicating whether there are nested columns in build/probe table
+ */
+// TODO: `HasNested` to be removed via dispatching
+template <has_nested HasNested = has_nested::NO>
+class unique_hash_join {
+ public:
+  unique_hash_join() = delete;
+  ~unique_hash_join();
+  unique_hash_join(unique_hash_join const&)            = delete;
+  unique_hash_join(unique_hash_join&&)                 = delete;
+  unique_hash_join& operator=(unique_hash_join const&) = delete;
+  unique_hash_join& operator=(unique_hash_join&&)      = delete;
+
+  /**
+   * @brief Constructs a unique hash join object for subsequent probe calls
+   *
+   * @param build The build table that doesn't contain duplicate elements
```

Comment:
Maybe phrase it the other way.
```suggestion
   * @param build The build table that contains unique elements
```